pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -202,6 +205,11 @@ public class PersistentTopic extends AbstractTopic implements Topic, AddEntryCal
     private CompletableFuture<Long> currentCompaction = CompletableFuture.completedFuture(COMPACTION_NEVER_RUN);
     private final CompactedTopic compactedTopic;
 
+    // TODO: Create compaction strategy from topic policy when exposing strategic compaction to users.
+    private static Map<String, TopicCompactionStrategy> strategicCompactionMap = Map.of(
+            ServiceUnitStateChannelImpl.TOPIC,
+            new ServiceUnitStateCompactionStrategy());
```

Comment:
Is it better to use `Optional<TopicCompactionStrategy> compactionStrategy` and set the value to `ServiceUnitStateCompactionStrategy` if the topic is `ServiceUnitStateChannelImpl.TOPIC`. Maintaining a map with the topic name as the key in a `PersistentTopic` instance is a little confusing.

Reply:
I am trying to avoid introducing a new member var in PersistentTopic as its memory unnecessarily multiplies as many as topics.

We can remove this hashmap and move each topic compaction strategy config to topic policy once Pulsar exposes this strategic compaction to users.

Reply:
Yes, maybe we can just add it to the topic policy now, the HierarchyTopicPolicies. Just do not expose any admin APIs.

Reply:
Regarding adding strategy in topic policy, I'm wondering that’s over-engineering at this point since we haven’t agreed to expose this to users. 

Also, another property in topic policy would multiply as many as topics.

Reply:
Ah ok, that makes sense to me.